src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,105 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int VALUE_FOR_MOVE = 1;
```

Comment:
The `VALUE_FOR_MOVE` constant is a bit unclear. A more descriptive name could be `ELEMENT_SHIFT_STEP` or similar to express its purpose in array manipulation methods.